evm/trie/database.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package trie
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/ChainSafe/log15"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/util"
+)
+
+type Database struct {
+	path string
+	db   *leveldb.DB
+	lock sync.RWMutex
+	log  log15.Logger
+}
+
+type ProofDatabase struct {
+	db   map[string][]byte
+	lock sync.RWMutex
+}
```

Comment:
please separate these into two files

Reply:
removed the Database as I ended up using the ethdb leveldb implementation